host/app/resources/file.ts

```diff
@@ -19,49 +22,76 @@ type FileResource =
       ready: true;
       content: string;
       name: string;
+      path: string;
       write(content: string): void;
     };
 
 class _FileResource extends Resource<Args> {
   private handle: FileSystemFileHandle | undefined;
   private lastModified: number | undefined;
   private interval: ReturnType<typeof setInterval>;
+  private _path: string | undefined;
   @tracked content: string | undefined;
   @tracked ready = false;
 
   constructor(owner: unknown, args: Args) {
     super(owner, args);
-    this.read(args.named.path, args.named.handle);
+    this.read(args.named.path, args.named.handle, args.named.onRedirect);
     this.interval = setInterval(
-      () => this.read(args.named.path, args.named.handle),
+      () =>
+        this.read(args.named.path, args.named.handle, args.named.onRedirect),
       1000
     );
     registerDestructor(this, () => clearInterval(this.interval));
   }
 
+  get path() {
+    return this._path;
+  }
+
   get name() {
     return this.handle?.name;
   }
 
   private async read(
     path: string | undefined,
-    dirHandle: FileSystemDirectoryHandle | undefined
+    dirHandle: FileSystemDirectoryHandle | undefined,
+    onRedirect: (newPath: string) => void
   ) {
     if (path && dirHandle) {
-      let { handle: subdir, filename } = await this.traverse(dirHandle, path);
+      let { handle: subdir, filename } = await traverse(dirHandle, path);
       let handle: FileSystemFileHandle | undefined;
       try {
         handle = await subdir.getFileHandle(filename);
       } catch (err: unknown) {
-        if ((err as DOMException).name === 'NotFoundError') {
-          console.error(`${path} was not found in the local realm`);
+        if ((err as DOMException).name !== 'NotFoundError') {
+          throw err;
+        }
+        if (!filename.includes('.')) {
+          // perform some basic module resolution
+          for (let extension of executableExtensions) {
+            try {
+              console.info(
+                `retrying with ${filename + extension} in the local realm`
+              );
+              handle = await subdir.getFileHandle(filename + extension);
+            } catch (innerErr: unknown) {
+              if ((err as DOMException).name !== 'NotFoundError') {
+                throw err;
+              }
+            }
+          }
+          if (handle) {
+            let pathSegments = path.split('/');
+            pathSegments.pop();
+            pathSegments.push(handle.name);
+            path = pathSegments.join('/');
+            onRedirect(path);
+          }
         }
-        throw err;
       }
       if (!handle) {
-        throw new Error(
-          `can't obtain file ${path} from the local realm, perhaps this is a directory?`
-        );
+        throw new Error(`can't obtain file ${path} from the local realm`);
```

Comment:
This was a really interesting situation I wasn't expecting. The imports for cards used to construct fields assume some module resolution--for instance, you might `import Person from './person'` in your card which resolves to the `./person.gts` file. Our ember routing needs to be smart enough to deal with some basic module resolution. I tackled this similar to how we dealt with it in the worker, where our file resource will retry common js-based extensions, and if it is able to failover it will execute a `onRedirect()` action callback which will change the route accordingly to reflect the resolved module's file name.